taskweaver/session/session.py

```diff
@@ -80,17 +83,27 @@ def __init__(
             },
         )
         self.session_injector.binder.bind(CodeExecutor, self.code_executor)
-        if self.config.code_gen_mode == "plugin_only":
-            self.code_interpreter = self.session_injector.get(CodeInterpreterPluginOnly)
-        elif self.config.code_gen_mode == "cli_only":
-            self.code_interpreter = self.session_injector.get(CodeInterpreterCLIOnly)
-        elif self.config.code_gen_mode == "python":
-            self.code_interpreter = self.session_injector.get(CodeInterpreter)
-        else:
-            raise ValueError(
-                f"Unknown code_gen_mode: {self.config.code_gen_mode}, "
-                f"only support 'plugin_only', 'cli_only', 'python'",
-            )
+
+        if len(self.config.roles) == 1 and self.config.roles[0] == "planner":
+            self.logger.info("Planner only mode enabled")
+
+        self.worker_instances = {}
+        import_modules_from_dir(
+            os.path.join(
+                self.config.src.module_base_path,
+                "ext_roles",
+            ),
+        )
+
+        for sub_cls in Role.__subclasses__():
+            if sub_cls is Planner:
+                continue
+            role_instance = self.session_injector.get(sub_cls)
+            if role_instance.name in self.config.roles:
+                self.worker_instances[role_instance.get_alias()] = role_instance
```

Comment:
do we need to not only load all classes of roles but also instanciate all of them them filter out what we need as specificed in config?